lib/rules/linebreak-style.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Rule to forbid mixing LF and LFCR line breaks.
+ * @author Erik Mueller
+ * @copyright 2015 Erik Mueller. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var linebreakStyle = context.options[0] || "unix";
+
+    return {
+        "Program": function checkForlinebreakStyle(node) {
+            var source = context.getSource(),
+                expectedLF = linebreakStyle === "unix",
+                // Assume LF if we find not one CRLF
+                isSolelyLF = source.indexOf("\r\n") === -1,
+                // Assume CRLF if we find no LF
+                isSolelyCRLF = source.match(/[^\r]\n/) === null;
+
+            if (expectedLF && isSolelyCRLF) {
+                context.report(node, "Expected linebreaks to be 'LF' but found 'CRLF'.");
+            } else if (!expectedLF && isSolelyLF) {
+                context.report(node, "Expected linebreaks to be 'CRLF' but found 'LF'.");
+            } else if (!isSolelyLF && !isSolelyCRLF) {
+                // NOTE How to tell the line where the error occured when only having the complete source code?
```

Comment:
Yeah see, I already noticed that ^^' But couldn't find a way to do it. The first two branches shouldn't be a problem since they are indicating that the whole line breaks are wrong. This one (line 30) is more interesting since the user seems to have mixed them.